pkg/generate/terraform_state.go

```diff
@@ -1,39 +1,16 @@
 package generate
 
 import (
-	"encoding/json"
+	"context"
 	"fmt"
-)
-
-type state map[string]interface{}
-type resource map[string]interface{}
-
-func (s state) resources() []resource {
-	values := s["values"].(map[string]interface{})
-	rootModule := values["root_module"].(map[string]interface{})
-	var resources []resource
-	for _, resourceInterface := range rootModule["resources"].([]interface{}) {
-		resources = append(resources, resourceInterface.(map[string]interface{}))
-	}
-	return resources
-}
 
-func (r resource) resourceType() string {
-	return r["type"].(string)
-}
-
-func (r resource) values() map[string]interface{} {
-	return r["values"].(map[string]interface{})
-}
+	tfjson "github.com/hashicorp/terraform-json"
+)
 
-func getState(dir string) (state, error) {
-	state, err := runTerraformWithOutput(dir, "show", "-json")
+func getState(ctx context.Context, cfg *Config) (*tfjson.State, error) {
```

Comment:
`tfexec` implements `tfjson` to read the state, hence the big changeset in this file.